### DISCUSSION TITLE
Cannot send transactions larger than 107309 bytes

### DISCUSSION THREAD

**Discussion by andyhass** [WalletConnect/walletconnect-monorepo#3045] [Developer Support]
Hi WalletConnect team - I wanted to start a thread on this issue: https://github.com/WalletConnect/walletconnect-monorepo/issues/3032

After upgrading to WC V2, users of my company's dApp started experiencing timeouts when using WC.  After a transaction larger than 107309 bytes was created (which is something WC1 could support without issues), the transaction would never appear in their wallet (e.g. Ledger Live, Fireblocks) and eventually an error without any details would be thrown from WC when the default 5 minute timeout for sending transactions was reached.  

In addition to this "large" transactions not being an issue with WalletConnect V1, it's not an issue with wallets (e.g. Metamask).  That said, WalletConnect V2 leads to a degraded application experience for our users.

**Comment by zlsgh** [comment]
We would greatly appreciate your escalation of this issue as it is currently limiting our customers from using WalletConnect to stake large amounts of ETH.

Due to the 32 ETH limit per validator with Ethereum staking, when a customer wants to stake a large amount we (along with other staking providers) batch them into one contract call, to a batch-deposit contract, which increases the data size sent through the WC relay. We are expecting a maximum of 455 KB. Can you increase the limit to above 455 KB?

Until this is resolved we have to direct our customers to avoid using WalletConnect for these transactions.

**Reply by Kirby-WalletConnect** [reply]
Hi @zlsgh confirming we are taking a look at this and should have fix this week with a limit increase for these types of requests. I'll send you an update once its confirmed in production.

**Comment by Kirby-WalletConnect** [comment]
Confirming we are taking a look at this and should have fix this week with a limit increase for these types of requests. I'll send an update once its confirmed in production.

**Reply by zlsgh** [reply]
Awesome. Thank you!

**Comment by Kirby-WalletConnect** [comment]
Limit increases should now be in production (about 10x increase and messaging if you happened to hit the limit). Let us know if you run into this again in the future and appreciate you bringing this to our attention.
